### PR TITLE
Upload @TudorMagirescu's UnsanitizedFilenameFromContentProvider lint check

### PR DESCRIPTION
**Pull request by bjiang7**
As discussed internally, @TudorMagirescu has written a lint check to flag when content provider file names are trusted without proper sanitization. Given that this lint check will be available in the stable version of Android Studio IDE directly in Jan 2025, we are allowing developers to "preview" the lint check here in this IDE-external lint library.

Great job @TudorMagirescu :)

### CODE REVIEW

**Comment by azhara-a** [comment]
LGTM, thanks!

File: app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    private fun useGetColumnIndexOrThrow(cursor: Cursor) {
+        val id = cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME)
+        val fileName = cursor.getString(id)
+
+        val fileObject = File("./", fileName)
+    }
+
+}
```
Comment (azhara-a) [inline app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt]: Nit: Can you add a new line at the end of the file please?

File: app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        val id = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
+        println(temp)
+        val fileName = cursor.getString(id)
+
+        temp += 2
+        val fileObject = File("./", fileName)
+    }
+
+    private fun callChainStartingAtContext(cursor: Cursor) {
```
Comment (TudorMagirescu) [inline app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt:27]: This method seems to be a duplicate of  `trustProvidedFilename_shouldWarn`.

**Comment by TudorMagirescu** [comment]
LGTM, but I have left a comment. Thanks for including `UnsafeFilenameDetector` in this library!

File: app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt (hunk 1 above)
Comment (bjiang7) [reply app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt]: Done!

File: app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt (hunk 2 above)
Comment (bjiang7) [reply app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt:27]: Thanks, I deleted trustProvidedFilename_shouldWarn! I had copied your first two test cases into FakeContentProviderFilenameSanitization as it was known code to trigger the lint firing, but didn't check that they were duplicates. Oops :)

File: app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt (hunk 1 above)
Comment (bjiang7) [reply app/src/main/java/com/android/example/FakeContentProviderFilenameSanitization.kt]: Hmm, I added a newline and it shows in Android Studio but it doesn't appear to have reflected it, so I've added two newlines now just to have the trailing newline show up properly :/

**Pull request merged**